widget/list.go

```diff
@@ -0,0 +1,431 @@
+		l.list.selectedItem = item.(*listItem)
+	}
+	if previousIndicator != item.(*listItem).selected {
+		item.Refresh()
+	}
+	if f := l.list.UpdateItem; f != nil {
+		f(index, item.(*listItem).child)
+	}
+	item.(*listItem).onTapped = func() {
+		if l.list.selectedItem != nil && l.list.selectedIndex >= l.firstItemIndex && l.list.selectedIndex <= l.lastItemIndex {
+			l.list.selectedItem.selected = false
+			l.list.selectedItem.Refresh()
+		}
+		l.list.selectedItem = item.(*listItem)
+		l.list.selectedIndex = index
+		l.list.selectedItem.selected = true
+		l.list.selectedItem.Refresh()
+		if f := l.list.OnItemSelected; f != nil {
+			f(index)
+		}
+	}
+}
+
+// Declare conformity with interfaces
+var _ fyne.Widget = (*listItem)(nil)
+var _ fyne.Tappable = (*listItem)(nil)
+var _ desktop.Hoverable = (*listItem)(nil)
+
+type listItem struct {
+	BaseWidget
+
+	onTapped          func()
+	statusIndicator   *canvas.Rectangle
+	child             fyne.CanvasObject
+	divider           *canvas.Rectangle
+	hovered, selected bool
+}
+
+func newListItem(child fyne.CanvasObject, tapped func()) *listItem {
+	li := &listItem{
+		child:    child,
+		onTapped: tapped,
+	}
+
+	li.ExtendBaseWidget(li)
+	return li
+}
+
+// CreateRenderer is a private method to Fyne which links this widget to its renderer
+func (li *listItem) CreateRenderer() fyne.WidgetRenderer {
+	li.ExtendBaseWidget(li)
+
+	li.statusIndicator = canvas.NewRectangle(theme.BackgroundColor())
+	li.divider = canvas.NewRectangle(theme.ShadowColor())
+
+	objects := []fyne.CanvasObject{li.statusIndicator, li.child, li.divider}
+
+	return &listItemRenderer{widget.NewBaseRenderer(objects), li}
+}
+
+// MinSize returns the size that this widget should not shrink below
+func (li *listItem) MinSize() fyne.Size {
+	li.ExtendBaseWidget(li)
+	return li.BaseWidget.MinSize()
+}
+
+// MouseIn is called when a desktop pointer enters the widget
+func (li *listItem) MouseIn(*desktop.MouseEvent) {
+	li.hovered = true
+	li.Refresh()
+}
+
+// MouseMoved is called when a desktop pointer hovers over the widget
+func (li *listItem) MouseMoved(*desktop.MouseEvent) {
+}
+
+// MouseOut is called when a desktop pointer exits the widget
+func (li *listItem) MouseOut() {
+	li.hovered = false
+	li.Refresh()
+}
+
+// Tapped is called when a pointer tapped event is captured and triggers any tap handler
+func (li *listItem) Tapped(*fyne.PointEvent) {
+	if li.onTapped != nil {
+		li.selected = true
+		li.Refresh()
+		li.onTapped()
+	}
+}
+
+// Declare conformity with the WidgetRenderer interface
+var _ fyne.WidgetRenderer = (*listItemRenderer)(nil)
+
+type listItemRenderer struct {
+	widget.BaseRenderer
+
+	item *listItem
+}
+
+// MinSize calculates the minimum size of a listItem.
+// This is based on the size of the status indicator and the size of the child object.
+func (li *listItemRenderer) MinSize() (size fyne.Size) {
+	itemSize := li.item.child.Size()
+	if itemSize.IsZero() {
+		itemSize = li.item.MinSize()
+	}
```

Comment:
I'm not sure this is the right fix. If this is called at the start and it gives a MinSize let's call `A`. Then the list is rendered in a large area, and each item is resized to be wider, then this function is called again it gives a MinSize let's call `B`. Then wont `B > A`? So this function is not returning "the minimum size of a listItem", it is returning "the size of a listItem".

Why not just have;
```go
itemSize := li.item.child.MinSize()
```

Reply:
Fair enough. Resolved.